src/ethernet.rs

```diff
@@ -81,84 +84,113 @@ struct VidEthertype {
+            0x8035 => Self::RARP,           // Reverse Address Resolution Protocol
+            0x809B => Self::AppleTalk,      // AppleTalk (Ethertalk)
+            0x80F3 => Self::AARP,           // AppleTalk Address Resolution Protocol (AARP)
+            0x8100 => Self::VLAN, // VLAN-tagged frame (IEEE 802.1Q) and Shortest Path Bridging IEEE 802.1aq[5]
+            0x8137 => Self::IPX,  // IPX
+            0x8204 => Self::Qnet, // QNX Qnet
+            0x86DD => Self::IPv6, // Internet Protocol Version 6 (IPv6)
+            0x8808 => Self::FlowControl, // Ethernet flow control
+            0x8819 => Self::CobraNet, // CobraNet
+            0x8847 => Self::MPLSuni, // MPLS unicast
+            0x8848 => Self::MPLSmulti, // MPLS multicast
+            0x8863 => Self::PPPoEdiscovery, // PPPoE Discovery Stage
+            0x8864 => Self::PPPoEsession, // PPPoE Session Stage
+            0x887B => Self::HomePlug, // HomePlug 1.0 MME
+            0x888E => Self::EAPOL, // EAP over LAN (IEEE 802.1X)
+            0x8892 => Self::PROFINET, // PROFINET Protocol
+            0x889A => Self::HyperSCSI, // HyperSCSI (SCSI over Ethernet)
+            0x88A2 => Self::ATAOE, // ATA over Ethernet
+            0x88A4 => Self::EtherCAT, // EtherCAT Protocol
+            0x88A8 => Self::QinQ, // Provider Bridging (IEEE 802.1ad) & Shortest Path Bridging IEEE 802.1aq[5]
+            0x88AB => Self::Powerlink, // Ethernet Powerlink[citation needed]
+            0x88B8 => Self::GOOSE, // GOOSE (Generic Object Oriented Substation event)
+            0x88B9 => Self::GSE,  // GSE (Generic Substation Events) Management Services
+            0x88CC => Self::LLDP, // Link Layer Discovery Protocol (LLDP)
+            0x88CD => Self::SERCOS, // SERCOS III
+            0x88E1 => Self::HomePlugAV, // HomePlug AV MME[citation needed]
+            0x88E3 => Self::MRP,  // Media Redundancy Protocol (IEC62439-2)
+            0x88E5 => Self::MACsec, // MAC security (IEEE 802.1AE)
+            0x88E7 => Self::PBB,  // Provider Backbone Bridges (PBB) (IEEE 802.1ah)
+            0x88F7 => Self::PTP,  // Precision Time Protocol (PTP) over Ethernet (IEEE 1588)
+            0x88FB => Self::PRP,  // Parallel Redundancy Protocol (PRP)
+            0x8902 => Self::CFM, // IEEE 802.1ag Connectivity Fault Management (CFM) Protocol / ITU-T Recommendation Y.1731 (OAM)
+            0x8906 => Self::FCoE, // Fibre Channel over Ethernet (FCoE)
+            0x8914 => Self::FCoEi, // FCoE Initialization Protocol
+            0x8915 => Self::RoCE, // RDMA over Converged Ethernet (RoCE)
+            0x891D => Self::TTE, // TTEthernet Protocol Control Frame (TTE)
+            0x892F => Self::HSR, // High-availability Seamless Redundancy (HSR)
+            0x9000 => Self::CTP, // Ethernet Configuration Testing Protocol[6]
+            0x9100 => Self::VLANdouble, // VLAN-tagged (IEEE 802.1Q) frame with double tagging
+            other => Self::Other(other),
+        }
     }
 }
 
+// to remove ?
```

Comment:
I think we should not make this function public, this is likely to be use only by us, in fact I would prefer inline this unwrap call.